enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,906 @@
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: X-Cache-Info
+              action:
+                type: Set
+                set:
+                  value: "not cacheable; meta data too large"
+            - name: X-XSS-Protection
+              action:
+                type: Delete
+            - name: X-Source
+              action:
+                type: Set
+                set:
+                  value: "%[res.hdr(X-Value),lower]"
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name: X-SSL-Client-Cert
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdr(host),lower]"
+```
+
+#### Variation [optional]
+
+### API Extensions
+
+A)  Setting or deleting HTTP request or response headers via `IngressController.spec`.
+The existing `IngressControllerHTTPHeaders` Go type definition of the IngressController API is extended by adding a field for header manipulations.
+```go
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type IngressControllerSpec struct {
+    // httpHeaders defines policy for HTTP headers.
+    //
+    // If this field is empty, the default values are used.
+    //
+    // +optional
+    HTTPHeaders *IngressControllerHTTPHeaders `json:"httpHeaders,omitempty"`
+}
+```
+
+```go
+// IngressControllerHTTPHeaders specifies how the IngressController handles
+// certain HTTP headers.
+type IngressControllerHTTPHeaders struct {
+    // actions specifies options for performing action on headers.
+    // Note that this option only applies to cleartext HTTP connections
+    // and to secure HTTP connections for which the ingress controller
+    // terminates encryption (that is, edge-terminated or reencrypt
+    // connections).  Headers cannot be set or deleted for TLS passthrough
+    // connections.
+    // Setting HSTS header is supported via another mechanism in Ingress.Spec.RequiredHSTSPolicies.
+    // Setting httpCaptureHeaders for the headers set via this API will not work.
+    // If spec.clientTLS, spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId, spec.httpHeaders.headerNameCaseAdjustments is set
+    // and if you set the same header via this API then that header will get replaced with the value specified via this API.
+    // If cache-control is set using this API then the existing value of cache-control will get replaced.
```

Comment:
The list of headers that are not allowed to be edited needs to be clearly listed, or added to the deny list so that users will be informed of what they are not allowed to edit.